### DISCUSSION TITLE
Add map of campus printer locations

### DISCUSSION THREAD

**Discussion by hannesmcman** [StoDevX/AAO-React-Native#5548] [Ideas]
This could either come in the shape of a homescreen button similar to campus map, or could go somewhere else, but I was basically imagining it as a simple image view containing the IT printer map (shown below). Thoughts? 
![image](https://user-images.githubusercontent.com/24640726/34631456-1257b2ae-f236-11e7-8257-75b78f6e472f.png)

**Comment by hawkrives** [comment]
@hannesmcman You might be interested in this branch: https://github.com/StoDevX/AAO-React-Native/tree/stoprint-take-2. I've been doing all the work I've shown off later in this thread on this branch.

**Comment by rye** [comment]
Part of the issue is that (a) this map is old (see: Dittman), and (b) this map doesn't have Holland.  We did a quick check of two people in IT and it doesn't seem that we have an updated map, yet?  Not sure, we'd have to look into this.

This might be a good thing to road-map with v3, which we hope might include mobile print release.

**Comment by drewvolz** [comment]
I'm not suggesting we go with this way, but we could try to collect some sort of structured data from the webpage's JSON [source](https://wp.stolaf.edu/it/wp-json/wp/v2/pages/3707).

**Comment by hawkrives** [comment]
Or we could just pull it from Papercut's JSON endpoints, https://papercut.stolaf.edu/client and such 😉

**Comment by drewvolz** [comment]
That's a much better idea.

**Comment by hawkrives** [comment]
<img width="375" alt="screen shot 2018-01-05 at 11 00 52 pm" src="https://user-images.githubusercontent.com/464441/34636657-59f1df96-f26c-11e7-9957-b2f66db144e4.png">

Progress!

**Comment by hawkrives** [comment]
My plan is to have the user flow be something like…

- StoPrint button opens the list of pending print jobs.
- The user can tap a print job to be taken to a list of printers, sorted by popularity (Papercut tells us this, I believe)
- The user can tap a printer to be alerted by a print confirmation dialog w/costs
- The user can confirm to release the print job on the chosen printer

I have two questions, I believe: 

1. Given that flow, is there any benefit to maintaining a separate screen of printers? Would it go printer -> jobs -> release job to printer?
2. If we stick with my initial proposal, what happens when the user has no pending print jobs? (Does the user need to see anything here? Maybe printing instructions? Or printer sustainability statistics?)

**Comment by hawkrives** [comment]
Note: If you want to test that branch, you must first log in to papercut's website, at https://papercut.stolaf.edu/client/. Only then will the app let you in.

Reasons why are still undergoing investigation.

**Comment by drewvolz** [comment]
Your initial proposal sounds like a better flow. They will most likely be in the StoPrint tab if they are trying to release a print job, so showing jobs first sounds ideal. If there are no jobs to be released, we could show a graphic and a message on a `<NoticeView />`.

On another note, we might receive a request to list "favorite" printers. So perhaps we can add that the same way @hannesmcman added it to building hours.

**Comment by hawkrives** [comment]
I think we can also do "recent printers", fwiw. I'm a-ok with favorites, too.

**Comment by hawkrives** [comment]
I've put off my working on the stoprint feature until post-2.5.
